```
Error running mint automation
+ mkdir -p 13189-a4a31db/gopath/src/github.com/minio
+ git -C 13189-a4a31db/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 13189-a4a31db/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 13189-a4a31db/gopath/src/github.com/minio/minio fetch --quiet minio pull/13189/head:pr13189
+ git -C 13189-a4a31db/gopath/src/github.com/minio/minio checkout --quiet pr13189
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/13189-a4a31db/gopath
+ make -C 13189-a4a31db/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/jcmturner/gokrb5/v8 v8.4.2
go: downloading github.com/minio/cli v1.22.0
go: downloading github.com/minio/pkg v1.1.2
go: downloading github.com/minio/minio-go/v7 v7.0.14-0.20210908194250-617d530ffac5
go: downloading github.com/colinmarc/hdfs/v2 v2.2.0
go: downloading cloud.google.com/go v0.65.0
go: downloading github.com/Azure/azure-storage-blob-go v0.10.0
go: downloading google.golang.org/api v0.31.0
go: downloading github.com/json-iterator/go v1.1.11
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading github.com/Azure/azure-pipeline-go v0.2.2
go: downloading github.com/minio/sha256-simd v1.0.0
go: downloading github.com/minio/madmin-go v1.1.5
go: downloading github.com/fatih/color v1.12.0
go: downloading github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: downloading github.com/golang-jwt/jwt v3.2.1+incompatible
go: downloading golang.org/x/crypto v0.0.0-20210616213533-5ff15b29337e
go: downloading github.com/minio/kes v0.14.0
go: downloading github.com/secure-io/sio-go v0.3.1
go: downloading golang.org/x/sys v0.0.0-20210630005230-0f9fa26af87c
go: downloading github.com/minio/sio v0.3.0
go: downloading github.com/minio/highwayhash v1.0.2
go: downloading github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: downloading github.com/Shopify/sarama v1.27.2
go: downloading github.com/VividCortex/ewma v1.1.1
go: downloading github.com/bcicen/jstream v1.0.1
go: downloading github.com/cheggaaa/pb v1.0.29
go: downloading github.com/bits-and-blooms/bloom/v3 v3.0.1
go: downloading github.com/klauspost/readahead v1.3.1
go: downloading github.com/klauspost/compress v1.13.5
go: downloading github.com/klauspost/reedsolomon v1.9.11
go: downloading github.com/djherbis/atime v1.0.0
go: downloading github.com/dchest/siphash v1.2.1
go: downloading github.com/go-openapi/loads v0.20.2
go: downloading github.com/cespare/xxhash v1.1.0
go: downloading github.com/gorilla/mux v1.8.0
go: downloading github.com/klauspost/pgzip v1.2.5
go: downloading github.com/google/uuid v1.1.2
go: downloading github.com/miekg/dns v1.1.35
go: downloading github.com/minio/console v0.9.8
go: downloading github.com/minio/selfupdate v0.3.1
go: downloading github.com/minio/zipindex v0.2.0
go: downloading github.com/philhofer/fwd v1.1.1
go: downloading github.com/pierrec/lz4 v2.6.0+incompatible
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/prometheus/client_golang v1.8.0
go: downloading github.com/prometheus/client_model v0.2.0
go: downloading github.com/prometheus/procfs v0.6.0
go: downloading github.com/rs/cors v1.7.0
go: downloading github.com/shirou/gopsutil v0.0.0-20190901111213-e4ec7b275ada
go: downloading github.com/tinylib/msgp v1.1.6-0.20210521143832-0becd170c402
go: downloading github.com/valyala/bytebufferpool v1.0.0
go: downloading go.etcd.io/etcd v0.5.0-alpha.5.0.20200910180754-dd1b699fc489
go: downloading go.uber.org/atomic v1.7.0
go: downloading github.com/beevik/ntp v0.3.0
go: downloading golang.org/x/time v0.0.0-20210220033141-f8bda1e9f3ba
go: downloading github.com/go-ldap/ldap v3.0.2+incompatible
go: downloading go.uber.org/zap v1.16.1-0.20210329175301-c23abee72d19
go: downloading github.com/cespare/xxhash/v2 v2.1.1
go: downloading github.com/coredns/coredns v1.4.0
go: downloading github.com/minio/simdjson-go v0.2.1
go: downloading github.com/montanaflynn/stats v0.6.6
go: downloading github.com/eclipse/paho.mqtt.golang v1.3.0
go: downloading github.com/ncw/directio v1.0.5
go: downloading github.com/elastic/go-elasticsearch/v7 v7.12.0
go: downloading github.com/go-sql-driver/mysql v1.5.0
go: downloading github.com/gomodule/redigo v2.0.0+incompatible
go: downloading github.com/lib/pq v1.9.0
go: downloading github.com/nats-io/nats.go v1.11.1-0.20210623165838-4b75fc59ae30
go: downloading github.com/nats-io/stan.go v0.8.3
go: downloading github.com/nsqio/go-nsq v1.0.8
go: downloading github.com/olivere/elastic/v7 v7.0.22
go: downloading github.com/go-ldap/ldap/v3 v3.2.4
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/streadway/amqp v1.0.0
go: downloading github.com/mattn/go-colorable v0.1.8
go: downloading github.com/mattn/go-isatty v0.0.13
go: downloading github.com/golang/protobuf v1.5.2
go: downloading github.com/mattn/go-ieproxy v0.0.1
go: downloading github.com/minio/csvparser v1.0.0
go: downloading github.com/minio/parquet-go v1.0.0
go: downloading github.com/xdg/stringprep v1.0.0
go: downloading github.com/alecthomas/participle v0.2.1
go: downloading github.com/shirou/gopsutil/v3 v3.21.7
go: downloading github.com/klauspost/cpuid v1.3.1
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/bits-and-blooms/bitset v1.2.0
go: downloading github.com/nats-io/nuid v1.0.1
go: downloading github.com/golang/snappy v0.0.3
go: downloading github.com/Azure/go-ntlmssp v0.0.0-20200615164410-66371956d46c
go: downloading github.com/go-asn1-ber/asn1-ber v1.5.1
go: downloading github.com/go-openapi/analysis v0.20.0
go: downloading github.com/go-openapi/spec v0.20.3
go: downloading github.com/go-openapi/swag v0.19.14
go: downloading github.com/mattn/go-runewidth v0.0.13
go: downloading github.com/prometheus/common v0.14.0
go: downloading github.com/beorn7/perks v1.0.1
go: downloading golang.org/x/net v0.0.0-20210421230115-4e50805a0758
go: downloading github.com/hashicorp/go-uuid v1.0.2
go: downloading github.com/jcmturner/gofork v1.0.0
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading github.com/jcmturner/dnsutils/v2 v2.0.0
go: downloading github.com/nats-io/nkeys v0.3.0
go: downloading golang.org/x/text v0.3.6
go: downloading github.com/minio/argon2 v1.0.0
go: downloading github.com/rivo/uniseg v0.2.0
go: downloading github.com/gorilla/websocket v1.4.2
go: downloading github.com/jcmturner/aescts/v2 v2.0.0
go: downloading github.com/mailru/easyjson v0.7.6
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading github.com/jcmturner/rpc/v2 v2.0.3
go: downloading google.golang.org/protobuf v1.26.0
go: downloading github.com/klauspost/cpuid/v2 v2.0.4
go: downloading git.apache.org/thrift.git v0.13.0
go: downloading go.uber.org/multierr v1.6.0
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.2-0.20181231171920-c182affec369
go: downloading github.com/tidwall/gjson v1.7.5
go: downloading github.com/tidwall/sjson v1.1.6
go: downloading github.com/lestrrat-go/jwx v1.2.0
go: downloading github.com/rjeczalik/notify v0.9.2
go: downloading github.com/fatih/structs v1.1.0
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/eapache/go-resiliency v1.2.0
go: downloading github.com/eapache/queue v1.1.0
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/rcrowley/go-metrics v0.0.0-20200313005456-10cdbea86bc0
go: downloading gopkg.in/jcmturner/gokrb5.v7 v7.5.0
go: downloading gopkg.in/ini.v1 v1.57.0
go: downloading github.com/minio/md5-simd v1.1.1
go: downloading github.com/josharian/intern v1.0.0
go: downloading github.com/rs/xid v1.2.1
go: downloading github.com/tklauser/go-sysconf v0.3.7
go: downloading github.com/go-openapi/strfmt v0.20.0
go: downloading github.com/go-openapi/jsonpointer v0.19.5
go: downloading github.com/go-openapi/jsonreference v0.19.5
go: downloading github.com/jcmturner/goidentity/v6 v6.0.1
go: downloading github.com/PuerkitoBio/purell v1.1.1
go: downloading github.com/tidwall/match v1.0.3
go: downloading github.com/tidwall/pretty v1.1.0
go: downloading github.com/PuerkitoBio/urlesc v0.0.0-20170810143723-de5bf2ad4578
go: downloading github.com/asaskevich/govalidator v0.0.0-20200907205600-7a23bdc65eef
go: downloading go.mongodb.org/mongo-driver v1.4.6
go: downloading github.com/go-openapi/errors v0.19.9
go: downloading github.com/mitchellh/mapstructure v1.4.1
go: downloading github.com/tklauser/numcpus v0.2.3
go: downloading github.com/lestrrat-go/option v1.0.0
go: downloading github.com/lestrrat-go/iter v1.0.1
go: downloading github.com/lestrrat-go/backoff/v2 v2.0.7
go: downloading github.com/lestrrat-go/httpcc v1.0.0
go: downloading github.com/lestrrat-go/blackmagic v1.0.0
go: downloading cloud.google.com/go/storage v1.10.0
go: downloading google.golang.org/genproto v0.0.0-20201110150050-8816d57aaa9a
go: downloading github.com/googleapis/gax-go v2.0.2+incompatible
go: downloading google.golang.org/grpc v1.37.0
go: downloading go.opencensus.io v0.22.5
go: downloading github.com/googleapis/gax-go/v2 v2.0.5
go: downloading github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e
go: downloading go.etcd.io/etcd/api/v3 v3.5.0-beta.4
go: downloading go.etcd.io/etcd/client/v3 v3.5.0-beta.4
go: downloading go.etcd.io/etcd/client/pkg/v3 v3.5.0-beta.4
go: downloading github.com/coreos/go-semver v0.3.0
go: downloading github.com/go-stack/stack v1.8.0
go: downloading github.com/coreos/go-systemd v0.0.0-20190719114852-fd7a80b32e1f
go: downloading github.com/coreos/go-systemd/v22 v22.3.1
go: downloading github.com/go-openapi/runtime v0.19.24
go: downloading github.com/minio/mc v0.0.0-20210626002108-cebf3318546f
go: downloading github.com/minio/operator v0.0.0-20210812082324-26350f153661
go: downloading github.com/unrolled/secure v1.0.7
go: downloading github.com/jessevdk/go-flags v1.4.0
go: downloading github.com/coreos/go-oidc v2.2.1+incompatible
go: downloading golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d
go: downloading k8s.io/client-go v0.21.1
go: downloading github.com/minio/direct-csi v1.3.5-0.20210601185811-f7776f7961bf
go: downloading github.com/go-openapi/validate v0.20.2
go: downloading github.com/pquerna/cachecontrol v0.0.0-20180517163645-1555304b9b35
go: downloading gopkg.in/square/go-jose.v2 v2.3.1
go: downloading github.com/docker/go-units v0.4.0
go: downloading k8s.io/apimachinery v0.21.1
go: downloading github.com/mb0/glob v0.0.0-20160210091149-1eb79d2de6c4
go: downloading k8s.io/kube-openapi v0.0.0-20210305001622-591a79e4bda7
go: downloading github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
go: downloading github.com/minio/operator/logsearchapi v0.0.0-20210812082324-26350f153661
go: downloading k8s.io/klog v1.0.0
go: downloading k8s.io/api v0.21.1
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading github.com/minio/filepath v1.0.0
go: downloading maze.io/x/duration v0.0.0-20160924141736-faac084b6075
go: downloading github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: downloading sigs.k8s.io/controller-runtime v0.8.0
go: downloading github.com/pkg/xattr v0.4.1
go: downloading github.com/pkg/profile v1.3.0
go: downloading github.com/minio/colorjson v1.0.1
go: downloading github.com/posener/complete v1.2.3
go: downloading gopkg.in/h2non/filetype.v1 v1.0.5
go: downloading golang.org/x/term v0.0.0-20210406210042-72f3dc4e9b72
go: downloading k8s.io/klog/v2 v2.8.0
go: downloading github.com/hashicorp/go-multierror v1.0.0
go: downloading github.com/hashicorp/errwrap v1.0.0
go: downloading github.com/go-logr/logr v0.4.0
go: downloading github.com/georgysavva/scany v0.2.7
go: downloading github.com/stretchr/testify v1.7.0
go: downloading sigs.k8s.io/yaml v1.2.0
go: downloading sigs.k8s.io/structured-merge-diff v1.0.1-0.20191108220359-b1b620dd3f06
go: downloading github.com/google/gofuzz v1.1.0
go: downloading github.com/google/go-cmp v0.5.5
go: downloading gopkg.in/inf.v0 v0.9.1
go: downloading github.com/pmezard/go-difflib v1.0.1-0.20181226105442-5d4384ee4fb2
go: downloading github.com/stretchr/objx v0.2.0
go: downloading gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
go: downloading github.com/emicklei/go-restful v2.9.5+incompatible
go: downloading sigs.k8s.io/structured-merge-diff/v4 v4.1.0
go: downloading k8s.io/utils v0.0.0-20201110183641-67b214c5f920
go: downloading github.com/googleapis/gnostic v0.5.1
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
# github.com/minio/minio/cmd/gateway/s3
cmd/gateway/s3/gateway-s3.go:38:2: imported and not used: "github.com/minio/minio/internal/config"
cmd/gateway/s3/gateway-s3.go:42:2: imported and not used: "github.com/minio/pkg/env"
cmd/gateway/s3/gateway-s3.go:206:6: undefined: serverDebugLog
make: *** [build] Error 2

```

--- a/cmd/gateway-main.go
+++ b/cmd/gateway-main.go
@@ -280,7 +280,6 @@ func StartGateway(ctx *cli.Context, gw Gateway) {
 		SecretKey: globalActiveCred.SecretKey,
 	})
 	if err != nil {
-		globalHTTPServer.Shutdown()
 		logger.FatalIf(err, "Unable to initialize gateway backend")
 	}
 	newObject = NewGatewayLayerWithLocker(newObject)

--- a/cmd/gateway/s3/gateway-s3.go
+++ b/cmd/gateway/s3/gateway-s3.go
@@ -97,12 +97,16 @@ func s3GatewayMain(ctx *cli.Context) {
 	logger.FatalIf(minio.ValidateGatewayArguments(serverAddr, args.First()), "Invalid argument")
 
 	// Start the gateway..
-	minio.StartGateway(ctx, &S3{args.First()})
+	minio.StartGateway(ctx, &S3{
+		host:  args.First(),
+		debug: env.Get("_MINIO_SERVER_DEBUG", config.EnableOff) == config.EnableOn,
+	})
 }
 
 // S3 implements Gateway.
 type S3 struct {
-	host string
+	host  string
+	debug bool
 }
 
 // Name implements Gateway interface.
@@ -203,10 +207,6 @@ func newS3(urlStr string, tripper http.RoundTripper) (*miniogo.Core, error) {
 		return nil, err
 	}
 
-	if env.Get("_MINIO_GATEWAY_DEBUG", config.EnableOff) == config.EnableOn {
-		clnt.TraceOn(os.Stderr)
-	}
-
 	return &miniogo.Core{Client: clnt}, nil
 }
 
@@ -224,6 +224,10 @@ func (g *S3) NewGatewayLayer(creds madmin.Credentials) (minio.ObjectLayer, error
 	clnt, err := newS3(g.host, t)
 	if err != nil {
 		return nil, err
+	}
+
+	if g.debug {
+		clnt.Client.TraceOn(os.Stderr)
 	}
 
 	probeBucketName := randString(60, rand.NewSource(time.Now().UnixNano()), "probe-bucket-sign-")

